src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,120 @@
+    private static RegistrationService registrationService;
+    private User defaultUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void before() {
+        defaultUser = new User();
+        defaultUser.setAge(DEFAULT_AGE);
+        defaultUser.setLogin(DEFAULT_LOGIN);
+        defaultUser.setPassword(DEFAULT_PASSWORD);
+    }
+
+    @AfterEach
+    void after() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_NullUser_notOk() {
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_alreadyExistedUser_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin(NON_UNIQUE_LOGIN);
+        Storage.people.add(existingUser);
+        defaultUser.setLogin(NON_UNIQUE_LOGIN);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        }, "method does not allow to store duplicates");
+    }
+
+    @Test
+    void register_passwordLength_notOk() {
+        defaultUser.setPassword("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        }, "method does not allow register user with shor passwords");
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        defaultUser.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        }, "password doesn`t have to be null");
+    }
+
+    @Test
+    void register_ageEquals18_ok() {
+        defaultUser.setAge(18);
+        assertEquals(defaultUser.getAge(),MIN_AGE);
+    }
+
+    @Test
+    void register_userLoginIsEmpty_notOk() {
+        defaultUser.setLogin("");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        });
+    }
+
+    @Test
+    void register_ageMoreThan130_notOk() {
+        defaultUser.setAge(131);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        });
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        defaultUser.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        });
+    }
+
+    @Test
+    void register_ageLessThan18_notOk() {
+        defaultUser.setAge(17);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(defaultUser);
+        });
+    }
+
+    @Test
+    void register_user_ok() {
+        registrationService.register(defaultUser);
+        assertEquals(1,defaultUser.getId());
+        assertEquals(DEFAULT_LOGIN,defaultUser.getLogin());
+        assertEquals(DEFAULT_PASSWORD,defaultUser.getPassword());
```

Comment:
```suggestion
        User registeredUser = registrationService.register(defaultUser);
        assertEquals(1, registeredUser.getId());
        assertEquals(defaultUser, registeredUser); // check all fields except id (see equals() in User class)
        // TODO: check if storage size increased
```

Reply:
Is it correct to create a new user ? i `ve already initialised it in beforeEach block.